WIP/Source/Veazy/src/main/java/vn/edu/fu/veazy/core/controller/ExamController.java

```diff
@@ -84,7 +96,166 @@ String createExam(@ModelAttribute("create-exam-form") CreateExamForm form,
+                if (isEquals) {
+                    rightAnswer++;
+                }
+                ExamAnswer answer = new ExamAnswer(exam, questionId, userAnswers, correctAnswers);
+                listQuestions.add(answer);
+            }
+            exam.setResult(rightAnswer / numberOfQuestion);
+            exam.setListQuestions(listQuestions);
+            //save in case is new exam
+            if (!form.getIsRedo()) {
+                examService.saveExam(exam);
+            }
+            GetExamResponseData data = new GetExamResponseData(exam);
+            response.setCode(ResponseCode.SUCCESS);
+            response.setData(data);
+            LOGGER.debug("submit exam successfully!");
+
+            return response.toResponseJson();
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+            LOGGER.error("Unknown error occured!");
+            response.setCode(ResponseCode.INTERNAL_SERVER_ERROR);
+        }
+        return response.toResponseJson();
+    }
+
+    /**
+     *
+     * @param examId
+     * @param principal
+     * @return json string
+     */
+    @PreAuthorize("!isAuthenticated()")
+    @RequestMapping(value = Const.URLMAPPING_GET_EXAM, method = RequestMethod.GET)
+    public @ResponseBody
+    String getExamAnswer(@PathVariable("exam_id") Integer examId,
+            Principal principal) {
+        Response response = new Response(ResponseCode.BAD_REQUEST);
+        try {
+            LOGGER.debug("Get to get exam answer controller successful");
+
+            String userName = principal.getName();
+            UserModel user = userService.findUserByUsername(userName);
+            if (user == null) {
+                LOGGER.debug("user not found!");
+                response.setCode(ResponseCode.USER_NOT_FOUND);
+                return response.toResponseJson();
+            }
+            ExamModel exam = examService.findExamById(examId);
+            if (!Objects.equals(user.getId(), exam.getUserId())) {
+                LOGGER.debug("user not allow!");
+                response.setCode(ResponseCode.USER_NOT_ALLOW);
+                return response.toResponseJson();
+            }
+            GetExamResponseData data = new GetExamResponseData(exam);
+            response.setCode(ResponseCode.SUCCESS);
+            response.setData(data);
+            LOGGER.debug("Get exam successfully!");
+
+            return response.toResponseJson();
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+            LOGGER.error("Unknown error occured!");
+            response.setCode(ResponseCode.INTERNAL_SERVER_ERROR);
+        }
+        return response.toResponseJson();
+    }
+
+    /**
+     *
+     * @param examId
+     * @param principal
+     * @return json string
+     */
+    @PreAuthorize("!isAuthenticated()")
+    @RequestMapping(value = Const.URLMAPPING_REDO_EXAM, method = RequestMethod.GET)
```

Comment:
@PreAuthorize("isAuthenticated()") // phai login thi moi redo duoc chu
or
@PreAuthorize("hasRole(3)") // cu the hon thi phai la learner (not editor || admin) moi dc redo